### PR TITLE
Support shadow colors using Figma variables

### LINKED ISSUE #1613: Shadow colors using Figma variables not supported
We currently just extract the raw color from the variable, so it never changes when variable modes or collections change dynamically.

### PR DESCRIPTION
**Pull request by rylin8**
Fixes #1613

### CODE REVIEW

**Comment by rylin8** [comment]
Not sure why the design switcher image shows a diff here, but the png looks correct.

**Comment by yiqunw700** [comment]
> Not sure why the design switcher image shows a diff here, but the png looks correct.

I am updating it. See #1628

**Comment by yiqunw700** [comment]
I'm wondering why some of the screenshots have a diff now, for example the battle ship?

File: designcompose/src/main/java/com/android/designcompose/FrameRender.kt [hunk 1]
```diff
@@ -810,7 +810,7 @@ internal fun squooshShapeRender(
 
         // Make an appropriate paint.
         val shadowPaint = Paint().asFrameworkPaint()
-        shadowPaint.color = shadow.shadowStyle.color.toColor().toArgb()
+        shadowPaint.color = shadow.shadowStyle.color.getValue(variableState)?.toArgb() ?: 0
```
Comment (yiqunw700) [inline designcompose/src/main/java/com/android/designcompose/FrameRender.kt:813]: I'm wondering if we should let it return@forEach when getValue(variableState) returns null

**Comment by rylin8** [comment]
> I'm wondering why some of the screenshots have a diff now, for example the battle ship?

You mean why I checked in new pngs? The roborazzi task made new images, but I don't see any visual diffs. @timothyfroehlich any idea why?

File: designcompose/src/main/java/com/android/designcompose/FrameRender.kt (hunk 1 above)
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/FrameRender.kt:813]: The forEach is for the list of shadow fills. If we fail to get the variable value out of one, I would think we should still keep going through the list to draw the others.

Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/FrameRender.kt:813]: Yes. I agree. 
`return@forEach` will skip the current one and continue the next one in the forEach loop, like a `continue` in Java?

Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/FrameRender.kt:813]: Oh right, was confused and thought it was like `break`.

Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/FrameRender.kt:814]: Done

File: designcompose/src/main/java/com/android/designcompose/FrameRender.kt [hunk 2]
```diff
@@ -547,7 +548,7 @@ internal fun ContentDrawScope.render(
...
```
Comment (yiqunw700) [inline designcompose/src/main/java/com/android/designcompose/FrameRender.kt:551]:
```suggestion
        shadowPaint.color = shadow.shadowStyle.color.getValue(variableState)?.toArgb() ?: return@forEach
```

File: designcompose/src/main/java/com/android/designcompose/FrameRender.kt [hunk 3]
```diff
@@ -810,7 +811,7 @@ internal fun squooshShapeRender(
...
```
Comment (yiqunw700) [inline designcompose/src/main/java/com/android/designcompose/FrameRender.kt:814]:
```suggestion
        shadowPaint.color = shadow.shadowStyle.color.getValue(variableState)?.toArgb() ?: return@forEach
```

File: designcompose/src/main/java/com/android/designcompose/FrameRender.kt [hunk 4]
```diff
@@ -869,7 +870,7 @@ internal fun squooshShapeRender(
...
```
Comment (yiqunw700) [inline designcompose/src/main/java/com/android/designcompose/FrameRender.kt:873]:
```suggestion
        shadowPaint.color = shadow.shadowStyle.color.getValue(variableState)?.toArgb() ?: return@forEach
```

**Comment by yiqunw700** [comment]
> > I'm wondering why some of the screenshots have a diff now, for example the battle ship?
> 
> You mean why I checked in new pngs? The roborazzi task made new images, but I don't see any visual diffs. @timothyfroehlich any idea why?

I  compared the the old battleship screenshots using the image diff tool, but can not tell anything different. It might just be that rgb data now is argb with extra alpha data but no visual difference.

**Review by yiqunw700 — APPROVED** [review]

File: designcompose/src/main/java/com/android/designcompose/FrameRender.kt (hunk 2 above)
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/FrameRender.kt]: Done

File: designcompose/src/main/java/com/android/designcompose/FrameRender.kt (hunk 3 above)
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/FrameRender.kt]: Done

File: designcompose/src/main/java/com/android/designcompose/FrameRender.kt (hunk 4 above)
Comment (rylin8) [reply designcompose/src/main/java/com/android/designcompose/FrameRender.kt]: Done

**Pull request merged**